scala-exercises-server/app/services/parser/ExercisecodeExtractor.scala

```diff
@@ -81,21 +104,52 @@ object ExerciseCodeExtractor {
       remaining match {
         case Nil ⇒ builder
         case h :: t ⇒
+
           val modBuilder = (h, builder.state) match {
             case (commentStart(startLine), DefiningClass) ⇒
               builder.addToCategoryDescription(startLine)
+            case (codeInComment(codeLine), DefiningClass) ⇒
```

Comment:
All of these changes are probably in direct conflict with the work @andyscott was doing in reworking the parser using the Scala compiler API's and we need a strategy before we consider merging this. @andyscott Should we merge your stuff first and try to add this stuff later?